core/src/main/java/com/linecorp/armeria/internal/common/CancellationScheduler.java

```diff
@@ -89,18 +111,18 @@ public void run(Throwable cause) { /* no-op */ }
 
     CompletableFuture<Throwable> whenCancelled();
 
-    @Deprecated
-    CompletableFuture<Void> whenTimingOut();
-
-    @Deprecated
-    CompletableFuture<Void> whenTimedOut();
+    /**
+     * Updates the task that will be executed once this scheduler completes either by the configured timeout,
+     * or immediately via {@link #finishNow()}. If the scheduler hasn't completed yet, the task will simply
+     * be updated. If the scheduler has already been triggered for completion, the supplied
+     * {@link CancellationTask} will be executed after the currently set task has finished executing.
+     */
+    void updateTask(CancellationTask cancellationTask);
 
     enum State {
         INIT,
-        INACTIVE,
-        SCHEDULED,
-        FINISHING,
-        FINISHED
+        PENDING,
```

Comment:
How about `SCHEDULED`? `PENDING` sounds a like a status between `init()` and `start()`.

Reply:
Fixed, note that technically calling `start()` doesn't guarantee a task has been scheduled if the timeout is `Long.MAX_VALUE`